signature/pipeline_invariants_test.go

```diff
@@ -0,0 +1,179 @@
+package signature
+
+import (
+	"testing"
+
+	"github.com/buildkite/go-pipeline"
+)
+
+func TestCommandStepWithInvariants_SignedFields_WithSecrets(t *testing.T) {
+	t.Parallel()
+
+	dbUrl := "DATABASE_URL"
+	step := commandStepWithInvariants{
+		CommandStep: pipeline.CommandStep{
+			Command: "echo hello",
+			Env:     map[string]string{"FOO": "bar"},
+			Plugins: pipeline.Plugins{},
+			Secrets: pipeline.Secrets{
+				{Key: "DATABASE_URL", EnvironmentVariable: &dbUrl},
+			},
+		},
+		RepositoryURL: "https://github.com/example/repo",
+		OuterEnv:      map[string]string{"PIPELINE_VAR": "value"},
+	}
+
+	fields, err := step.SignedFields()
+	if err != nil {
+		t.Fatalf("SignedFields() error = %v", err)
```

Comment:
[Optional thoughts] Perhaps we should make clear that `SignedFields` is a method of `step`, e.g.

```suggestion
		t.Fatalf("step.SignedFields() error = %v", err)
```

and maybe even that `step` has the type `commandStepWithInvariants`, but there's probably enough context to figure that out already.